libcxx/utils/generate_feature_test_macro_components.py

```diff
@@ -1857,6 +1858,38 @@ def produce_docs():
         f.write(doc_str)
 
 
+def get_table(data):
```

Comment:
Can you document the format of the data taken by this function? Also, the script assumes a lot of global state right now (like `get_std_dialects()`), it would be nice to improve on this as we're basically rewriting this. Can't we actually infer the standard dialects from the json data we're reading?

We could potentially have a Python class that represents this JSON data and provides a few useful attributes like `.dialects` and more, and then we could query the things we implement with something like

```python
for std in table.dialects:
  for ftm in table.entries:
    if ftm.implemented:
      # print information for this FTM in this dialect
```

Reply:
I disagree with the "a lot of global state". It only requires `get_std_dialects()`.

I've looked at your Python class suggestion and extracting the C++ dialects from the data. This indeed seems to work and we can indeed extract the dialects from the data.

I've updated the documentation and added more tests; the next version of the patch no longer is considered RFC.